packages/core/src/__tests__/state-manager.test.ts

```diff
@@ -165,12 +165,14 @@ test('handleTip for commit already in log', async () => {
   await (ceramic2.repository.stateManager as any)._handleTip(streamState2, stream1.state.log[1].cid)
 
   expect(streamState2.state).toEqual(stream1.state)
-  expect(retrieveCommitSpy).toBeCalledTimes(3) // TODO(1421): This should be 2!
+  // 4 IPFS retrievals - 2 each (signed commit and linked commit) for CID of commit to be applied and CID of genesis
+  // commit already in the stream state.
+  expect(retrieveCommitSpy).toBeCalledTimes(4)
```

Comment:
@smrz2001 I am confused. The goal is to reduce calls, but now it seems we have one more call. Is it really an optimisation? Or, maybe I misunderstand something?

Reply:
No, you're absolutely right.

The reason the number of calls went up here is because the previous count was incorrect.

In the previous version of the code, there was a direct call to `ipfs.dag.get` that was not being counted. I have now consolidated all IPFS lookups in `conflictResolution.ts` as well as both document handlers and 4 is the correct (and lowest) count.